internal/events/eventhandlers/send_receiver_wallets_sms_invitation_event_handler.go

```diff
@@ -0,0 +1,108 @@
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/crashtracker"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/data"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/events"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/events/schemas"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/message"
+	"github.com/stellar/stellar-disbursement-platform-backend/internal/services"
+	"github.com/stellar/stellar-disbursement-platform-backend/stellar-multitenant/pkg/tenant"
+)
+
+type SendReceiverWalletsSMSInvitationEventHandlerOptions struct {
+	DBConnectionPool               db.DBConnectionPool
+	AnchorPlatformBaseSepURL       string
+	MessengerClient                message.MessengerClient
+	MaxInvitationSMSResendAttempts int64
+	Sep10SigningPrivateKey         string
+	CrashTrackerClient             crashtracker.CrashTrackerClient
+}
+
+type SendReceiverWalletsSMSInvitationEventHandler struct {
+	tenantManager      tenant.ManagerInterface
+	crashTrackerClient crashtracker.CrashTrackerClient
+	service            services.SendReceiverWalletInviteServiceInterface
+}
+
+var _ events.EventHandler = new(SendReceiverWalletsSMSInvitationEventHandler)
+
+func NewSendReceiverWalletsSMSInvitationEventHandler(options SendReceiverWalletsSMSInvitationEventHandlerOptions) *SendReceiverWalletsSMSInvitationEventHandler {
+	s, err := services.NewSendReceiverWalletInviteService(
+		nil,
+		options.MessengerClient,
+		options.AnchorPlatformBaseSepURL,
+		options.Sep10SigningPrivateKey,
+		options.MaxInvitationSMSResendAttempts,
+		options.CrashTrackerClient,
+	)
+	if err != nil {
+		log.Fatalf("error instantiating service: %s", err.Error())
+	}
+
+	return &SendReceiverWalletsSMSInvitationEventHandler{
+		tenantManager:      tenant.NewManager(tenant.WithDatabase(options.DBConnectionPool)),
+		service:            s,
+		crashTrackerClient: options.CrashTrackerClient,
+	}
+}
+
+func (h *SendReceiverWalletsSMSInvitationEventHandler) Name() string {
+	return "SendReceiverWalletsSMSInvitationEventHandler"
+}
+
+func (h *SendReceiverWalletsSMSInvitationEventHandler) CanHandleMessage(ctx context.Context, message *events.Message) bool {
+	return message.Topic == events.ReceiverWalletSMSInvitationTopic
+}
+
+func (h *SendReceiverWalletsSMSInvitationEventHandler) Handle(ctx context.Context, message *events.Message) {
+	dataJSON, err := json.Marshal(message.Data)
+	if err != nil {
+		h.crashTrackerClient.LogAndReportErrors(ctx, err, fmt.Sprintf("[SendReceiverWalletsSMSInvitationEventHandler] could not marshal data: %v", message.Data))
+		return
+	}
+
+	var receiverWalletInvitationData []schemas.EventReceiverWalletSMSInvitationData
+	err = json.Unmarshal(dataJSON, &receiverWalletInvitationData)
+	if err != nil {
+		h.crashTrackerClient.LogAndReportErrors(ctx, err, fmt.Sprintf("[SendReceiverWalletsSMSInvitationEventHandler] could not unmarshal data: %v", message.Data))
+		return
+	}
```

Comment:
Unfortunately, we can't do this because the `message.Data` will be a `map` type and not a `[]schemas.EventReceiverWalletSMSInvitationData` pointer. Does that make sense?

So if we marshal and unmarshal a `map` we get the expected value.